From 9a6f74102c35e60f0f9298d417593f6cf4baa3c8 Mon Sep 17 00:00:00 2001
From: Kondapally Kalyan <kalyan.kondapally@intel.com>
Date: Fri, 4 Oct 2013 11:39:08 +0300
Subject: [PATCH] Chromium-Add-gl-changes-for-Wayland.

This patch adds basic support needed in NativeViewGLSurfaceEGL to resize window.
Changes related to resizing needs more testing before trying to upstream the patch.
---
 ui/gl/gl_surface_egl.cc |   13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/ui/gl/gl_surface_egl.cc b/ui/gl/gl_surface_egl.cc
index 0c1871e..9e5f7f5 100644
--- a/ui/gl/gl_surface_egl.cc
+++ b/ui/gl/gl_surface_egl.cc
@@ -116,8 +116,7 @@ bool GLSurfaceEGL::InitializeOneOff() {
     LOG(ERROR) << "OZONE failed to initialize hardware";
     return false;
   }
-  g_native_display = reinterpret_cast<EGLNativeDisplayType>(
-      surface_factory->GetNativeDisplay());
+  g_native_display = surface_factory->GetNativeDisplay();
 #else
   g_native_display = EGL_DEFAULT_DISPLAY;
 #endif
@@ -394,6 +393,7 @@ bool NativeViewGLSurfaceEGL::Resize(const gfx::Size& size) {
   GLContext* current_context = GLContext::GetCurrent();
   bool was_current =
       current_context && current_context->IsCurrent(this);
+
   if (was_current) {
     scoped_make_current.reset(
         new ui::ScopedMakeCurrent(current_context, this));
@@ -402,6 +402,15 @@ bool NativeViewGLSurfaceEGL::Resize(const gfx::Size& size) {
 
   Destroy();
 
+#if defined(USE_OZONE)
+  gfx::SurfaceFactoryOzone* surface_factory =
+    gfx::SurfaceFactoryOzone::GetInstance();
+
+  surface_factory->AttemptToResizeAcceleratedWidget(window_,
+                                                    gfx::Rect(0, 0, size.width(),
+                                                              size.height()));
+#endif
+
   if (!Initialize()) {
     LOG(ERROR) << "Failed to resize window.";
     return false;
-- 
1.7.9.5

